init.lua

```diff
@@ -37,57 +31,129 @@ local config = {
+        end
+    else
+        return 'nil'
     end
 end
 
-local function save_config()
-    local ini = {Settings = {}}
+---@param path string
+---@param tbl table
+local function pickle_save(path, tbl)
+    local tmp = path .. '.tmp'
+    local f, err = io.open(tmp, 'wb')
+    if not f then error('pickle save open failed: '..tostring(err)) end
+    f:write('return ')
+    f:write(serialize_lua(tbl))
+    f:write('\n')
+    f:close()
+    os.remove(path) -- ignore result
+    local ok, rerr = os.rename(tmp, path)
+    if not ok then error('pickle rename failed: '..tostring(rerr)) end
+end
 
-    ini.Settings.arrow_scale = string.format('%.2f', tonumber(config.arrow_scale) or 4.0)
-    ini.Settings.color_r     = string.format('%.2f', tonumber(config.arrow_color[1]) or 1.0)
-    ini.Settings.color_g     = string.format('%.2f', tonumber(config.arrow_color[2]) or 1.0)
-    ini.Settings.color_b     = string.format('%.2f', tonumber(config.arrow_color[3]) or 0.4)
-    ini.Settings.color_a     = string.format('%.2f', tonumber(config.arrow_color[4]) or 1.0)
+---@param path string
+---@return table|nil
+local function pickle_load(path)
+    if not file_exists(path) then return nil end
+    local ok, res = pcall(dofile, path)
+    if ok and type(res) == 'table' then return res end
+    return nil
+end
 
-    ini.Settings.window_x    = tostring(math.floor(tonumber(config.window_x or -1)))
-    ini.Settings.window_y    = tostring(math.floor(tonumber(config.window_y or -1)))
-    ini.Settings.window_w    = tostring(math.floor(tonumber(config.window_w or 160)))
-    ini.Settings.window_h    = tostring(math.floor(tonumber(config.window_h or 220)))
+---@return nil
+local function load_config()
+    local path = get_config_path()
+    local saved = pickle_load(path)
+    if not saved then return end
+    for k,v in pairs(saved) do
+        config[k] = v
+    end
+end
 
-    local ok, err = pcall(LIP.save, config_file, ini)
+---@return nil
+local function save_config()
+    local path = get_config_path()
+    local to_save = {
+        arrow_scale = tonumber(config.arrow_scale) or 4.0,
+        arrow_color = {
+            tonumber(config.arrow_color[1]) or 1.0,
+            tonumber(config.arrow_color[2]) or 1.0,
+            tonumber(config.arrow_color[3]) or 0.4,
+            tonumber(config.arrow_color[4]) or 1.0,
+        },
+        window_x = tonumber(config.window_x) or -1,
+        window_y = tonumber(config.window_y) or -1,
+        window_w = tonumber(config.window_w) or 160,
+        window_h = tonumber(config.window_h) or 220,
+        schema = 1,
+    }
+    local ok, err = pcall(pickle_save, path, to_save)
     if not ok then
         print(('\ar[ARROW save error]\ax %s'):format(tostring(err)))
     end
 end
 
 ---@param draw_list userdata
----@param heading number
----@param size number
----@param corner ImVec2
----@param color integer
+---@param heading number       -- degrees

```

Comment:
```suggestion
---@param heading number degrees

```

everything after the type is the description, the additional whitepsace and formatting is unnecessary fyi